### PR TITLE
ICU-20436 Add getDefaultHourCycle to DateTimePatternGenerator

### PR DESCRIPTION
**Pull request by FrankYFTang**
##### Checklist

- [x] Issue filed: https://unicode-org.atlassian.net/browse/ICU-20436
- [x] Updated PR title and link in previous line to include Issue number
- [x] Issue accepted
- [x] Tests included
- [x] Documentation is changed or added

### CODE REVIEW

File: icu4c/source/i18n/unicode/udatpg.h [hunk 1]
```diff
@@ -651,4 +669,14 @@ udatpg_getPatternForSkeleton(const UDateTimePatternGenerator *dtpg,
...
 
+/**
+ * Return the default hour cycle.
+ *
+ * @param dtpg a pointer to UDateTimePatternGenerator.
+ * @return the default hour cycle.
+ * @stable ICU 67
+ */
+U_DRAFT UDateTimeHourCycle U_EXPORT2
+udatpg_getDefaultHourCycle(const UDateTimePatternGenerator *dtpg);
```
Comment (sffc) [inline icu4c/source/i18n/unicode/udatpg.h]: I'd make this take an error code, even if you don't need to use it right now.  If the code backing this API were refactored in the future in a way that would need the error code, it's good to have it.

File: icu4j/main/classes/core/src/com/ibm/icu/text/DateTimePatternGenerator.java [hunk 2]
```diff
@@ -1205,6 +1205,35 @@ private String cldrKey() {
...
+    /**
+     * Hour Cycle
+     * @draft ICU 67
+     */
+    public enum HourCycle {
+        /**
+         * hour in am/pm (1~11)
+         * @draft ICU 67
+         */
+        HourCycle11,
```
Comment (sffc) [inline icu4j/main/classes/core/src/com/ibm/icu/text/DateTimePatternGenerator.java]: You should use UPPER_CONSTANT_CASE for enums

File: icu4c/source/i18n/unicode/udatpg.h (hunk 1 above)
Comment (sffc) [reply icu4c/source/i18n/unicode/udatpg.h]: The API proposal was approved this morning without the error code.

@FrankYFTang Can you send an email to icu-core asking if we should add the error code?

Comment (FrankYFTang) [reply icu4c/source/i18n/unicode/udatpg.h]: sorry, I forgot about that.

File: icu4c/source/i18n/unicode/udatpg.h [hunk 3]
```diff
@@ -651,4 +652,18 @@ udatpg_getPatternForSkeleton(const UDateTimePatternGenerator *dtpg,
...
 
+#ifndef U_HIDE_DRAFT_API
+/**
+ * Return the default hour cycle.
+ *
+ * @param dtpg a pointer to UDateTimePatternGenerator.
+ * @param pErrorCode a pointer to the UErrorCode which must not indicate a
+ *                   failure before the function call.
+ * @return the default hour cycle.
+ * @stable ICU 67
```
Comment (sffc) [inline icu4c/source/i18n/unicode/udatpg.h]: `@draft`

File: icu4c/source/i18n/unicode/udat.h [hunk 4]
```diff
@@ -958,7 +958,25 @@ udat_getBooleanAttribute(const UDateFormat* fmt, UDateFormatBooleanAttribute att
 U_CAPI void U_EXPORT2
 udat_setBooleanAttribute(UDateFormat *fmt, UDateFormatBooleanAttribute attr, UBool newValue, UErrorCode* status);
 
+#ifndef U_HIDE_DRAFT_API
+/**
+ * Hour Cycle.
+ * @draft ICU 67
+ */
+typedef enum UDateFormatHourCycle {
+    /* hour in am/pm (0~11) */
```
Comment (sffc) [inline icu4c/source/i18n/unicode/udat.h]: Please make these `/**` docs and use `@draft ICU 67` on each one, like you did in Java

File: icu4j/main/classes/core/src/com/ibm/icu/text/DateTimePatternGenerator.java (hunk 2 above)
Comment (FrankYFTang) [reply icu4j/main/classes/core/src/com/ibm/icu/text/DateTimePatternGenerator.java]: DONE

File: icu4c/source/i18n/unicode/udatpg.h (hunk 3 above)
Comment (FrankYFTang) [reply icu4c/source/i18n/unicode/udatpg.h]: DONE

File: icu4c/source/i18n/unicode/udat.h (hunk 4 above)
Comment (FrankYFTang) [reply icu4c/source/i18n/unicode/udat.h]: DONE

**Pull request merged**